File: enode/src/main/java/org/enodeframework/messaging/IMessageHandlerProxy3.java
package org.enodeframework.messaging;

import org.enodeframework.common.io.AsyncTaskResult;
import org.enodeframework.infrastructure.IObjectProxy;
import org.enodeframework.infrastructure.MethodInvocation;

import java.util.concurrent.CompletableFuture;

public interface IMessageHandlerProxy3 extends IObjectProxy, MethodInvocation {
    CompletableFuture<AsyncTaskResult> handleAsync(IMessage message1, IMessage message2, IMessage message3);
}

File: enode/src/main/java/org/enodeframework/messaging/IMessageHandlerProxy3.kt
package org.enodeframework.messaging

import org.enodeframework.infrastructure.IObjectProxy
import org.enodeframework.infrastructure.MethodInvocation

interface IMessageHandlerProxy3 : IObjectProxy, MethodInvocation {
    suspend fun handleAsync(message1: IMessage, message2: IMessage, message3: IMessage)
}